### ISSUE TITLE
Make Debian directory structure consistent

### ISSUE THREAD

**Issue by mfeit-internet2** [perfsonar/pscheduler#787]
Some pScheduler packages have a directory layout that varies from the convention of placing the Debian-related files under `pscheduler/PACKAGE-NAME/debian` and are instead using `pscheduler/PACKAGE-NAME/PACKAGE-NAME/debian`.

The affected packages are:

 * drop-in
 * jq
 * pscheduler-core
 * pscheduler-docs
 * pscheduler-jq-library
 * pscheduler-server
 * python-detach
 * python-functools32
 * python-icmperror
 * python-isodate
 * python-jsonschema
 * python-jsontemplate
 * python-pscheduler
 * python-psycopg2
 * python-pyjq
 * python-subprocess32
 * python-tzlocal

**Comment by vvidic** [comment]
On Fri, Mar 15, 2019 at 05:58:28AM -0700, Mark Feit wrote:
> Some pScheduler packages have a directory layout that varies from the
> convention of placing the Debian-related files under
> `pscheduler/PACKAGE-NAME/debian` and are instead using
> `pscheduler/PACKAGE-NAME/PACKAGE-NAME/debian`.

Right, but we can't have debian dir on the same level as
python-tzlocal_1.2.2.orig.tar.gz. dpkg-buildpackage
expects to find the source in .. directory so we have:

python-tzlocal/debian
python-tzlocal_1.2.2.orig.tar.gz

-- 
Valentin Vidic
Computer Systems Engineer - Expert
Department of Computer Infrastructure and Services
Croatian Academic and Research Network - CARNET
Josipa Marohnica 5, HR-10000 Zagreb, Croatia
tel: +385 1 6661 714, fax. +385 1 6661 635
gsm: +385 91 2480 919
www.CARNet.hr

**Comment by mfeit-internet2** [comment]
Well, that throws a wrench into the works.

Is there anything we can do with symlinks to keep the Debian stuff contained and not force having duplicates of the source tarballs?

**Comment by vvidic** [comment]
On Fri, Mar 15, 2019 at 08:52:44AM -0700, Mark Feit wrote:
> Well, that throws a wrench into the works.
> 
> Is there anything we can do with symlinks to keep the Debian stuff
> contained and not force having duplicates of the source tarballs?

I think we had some problems with the symlinks but we can try again.

But also if the files are the same git will store them only once so
it does not do any harm except it does not look nice.

-- 
Valentin Vidic
Computer Systems Engineer - Expert
Department of Computer Infrastructure and Services
Croatian Academic and Research Network - CARNET
Josipa Marohnica 5, HR-10000 Zagreb, Croatia
tel: +385 1 6661 714, fax. +385 1 6661 635
gsm: +385 91 2480 919
www.CARNet.hr

**Comment by vvidic** [comment]
It seems symlinks cause problems for Jenkins:

```
cp: cannot stat `/tmp/temp-lintian-lab-VghFbCr_SJ/pool/p/python-jsonschema/python-jsonschema_3.0.1-1~20190319091856_source/python-jsonschema_3.0.1.orig.tar.gz': No such file or directory
```
The symlink gets copied to /tmp but not the destination file. This is why we used the real file for orig.tar.gz.

**Comment by mfeit-internet2** [comment]
Well, then, do what you must.

**Comment by vvidic** [comment]
Yes, guess there is not much we can change than so close?
